Update compability of Octopus.Client/octo.exe for versions upto and including 2018.12

hnrkndrssn opened the pull request:
https://trello.com/c/jBuW0lx6/282-compatibility-matrix-in-docs-is-incorrect

wordlee commented:
LGTM!

matt-richardson commented:
Octopus clients 6.0 is out now, which contains a breaking change for 2019.2.

```
| 2019.1           | 5.x                   | 3.5 ➜ latest    | 4.0 ➜ latest | coming soon      |
| 2019.2 ➜ latest | 6.0.0 ➜ latest         | 3.5 ➜ latest    | 4.0 ➜ latest | coming soon      |
```

also think a lot of the `➜ latest` aren't actually correct

hnrkndrssn commented:
> Octopus clients 6.0 is out now, which contains a breaking change for 2019.2.
> 
> ```
> | 2019.1           | 5.x                   | 3.5 ➜ latest    | 4.0 ➜ latest | coming soon      |
> | 2019.2 ➜ latest | 6.0.0 ➜ latest         | 3.5 ➜ latest    | 4.0 ➜ latest | coming soon      |
> ```
> also think a lot of the `➜ latest` aren't actually correct

Added and updated
